test_unstructured/partition/docx/test_docx.py

```diff
@@ -764,6 +764,32 @@ def test_partition_docx_includes_hyperlink_metadata():
     assert metadata.link_urls is None
 
 
+# -- shape behaviors -----------------------------------------------------------------------------
+
+
+def test_include_inline_shapes_text():
+    elements = partition_docx(example_doc_path("docx-shapes.docx"))
+
+    assert len(elements) == 3
+
+    paragraph1 = elements[0]
+    paragraph2 = elements[1]
+
+    # <bracketed> text is written inside inline shapes
+    assert paragraph1.text == "Paragraph with single <inline-image> within."
+    assert paragraph2.text == "Paragraph with <inline-image1> and <inline-image2> within."
+
+
+def test_exclude_floating_shapes_text():
+    elements = partition_docx(example_doc_path("docx-shapes.docx"))
+
+    assert len(elements) == 3
+
+    paragraph = elements[2]
+
+    assert paragraph.text == "Paragraph with floating shape attached."
```

Comment:
Okay, now I think this simplifies to:
```python
def test_it_considers_text_inside_shapes():
    # -- <bracketed> text is written inside inline shapes --
    assert partition_docx(example_doc_path("docx-shapes.docx")) == [
        NarrativeText("Paragraph with single <inline-image> within.")
        NarrativeText("Paragraph with <inline-image1> and <inline-image2> within.")
        # -- text "<floating-shape>" in floating shape is ignored --
        NarrativeText("Paragraph with floating shape attached.")
    ]
```

You might need to adjust the Element sub-types to match, they might not be classified as NarrativeText.

Reply:
I don't now if this is necessarily better.

1. Regardless of how these paragraphs get classified (e.g. here first two are `Title`, third one is `NarrativeText`) their text should remain the same, so looking at text directly avoids test breaking on unrelated change.
2. If this detection breaks and text is different `pytest -vv` will tell you only that the objects differ like this:
  ```
  E         -     <unstructured.documents.elements.Title object at 0x7f2b326af520>,
  E         ?                                                              ^^ ^^
  E         +     <unstructured.documents.elements.Title object at 0x7f2b324f3040>,
  ```
which isn't very helpful until you unpack it.

Reply:
You can avoid coupling to the exact element type by changing it to:
```python
assert [element.text for element in partition_docx(...)] == [
    "string 1",
    "string 2",
    "string 3",
]
```
That's probably not a bad idea since this particular behavior should have no direct influence on which element the paragraph is classified as.

Reply:
Alright, I merged these tests and we're checking only text content.